code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/FeedItemSchemaData.java

```diff
@@ -0,0 +1,108 @@
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.messaging;
+
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.CONTENT;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.CONTENT_TYPE;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.EXPIRY_DATE;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.METADATA;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailDataKeys.PUBLISHED_DATE;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.ConsequenceDetailKeys.DATA;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.LOG_TAG;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.MessageFeedKeys.ACTION_TITLE;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.MessageFeedKeys.ACTION_URL;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.MessageFeedKeys.BODY;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.MessageFeedKeys.IMAGE_URL;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.MessageFeedKeys.TITLE;
+
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.util.DataReader;
+import com.adobe.marketing.mobile.util.DataReaderException;
+import com.adobe.marketing.mobile.util.JSONUtils;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Collections;
+import java.util.Map;
+
+// represents the schema data object for a feed item schema
+public class FeedItemSchemaData {
+    private static final String SELF_TAG = "FeedItemSchemaData";
+    private Object content;
+    private ContentType contentType;
+    private int publishedDate;
+    private int expiryDate;
+    private Map<String, Object> meta;
+
+    FeedItemSchemaData(final JSONObject schemaData) {
+        try {
+            final JSONObject consequenceDetails = InternalMessagingUtils.getConsequenceDetails(schemaData);
+            final Map<String, Object> feedItemData = JSONUtils.toMap(consequenceDetails.getJSONObject(DATA));
+            this.contentType = ContentType.fromString(DataReader.getString(feedItemData, CONTENT_TYPE));
+            if (contentType.equals(ContentType.APPLICATION_JSON)) {
+                this.content = DataReader.getTypedMap(Object.class, feedItemData, CONTENT);
```

Comment:
Is this always guaranteed to be a `JSONObject`? Any chance we might get `JSONArray`?

Reply:
Will the feed UI allows for custom content or will it be only OOTB Feed Item?

Reply:
in-app and feed items are always delivered via a ruleset schema which each rule being a JSON object. For now only OOTB feed items are supported.